File: Backend/Car_Rental_Spring/src/main/java/com/shriram/Car_Rental_Spring/services/customer/CustomerService.java
package com.shriram.Car_Rental_Spring.services.customer;

import com.shriram.Car_Rental_Spring.dto.BookACarDto;
import com.shriram.Car_Rental_Spring.dto.CarDto;
import com.shriram.Car_Rental_Spring.dto.CarDtoListDto;
import com.shriram.Car_Rental_Spring.dto.SearchCarDto;

import java.util.List;

public interface CustomerService {

    List<CarDto> getAllCars();

    boolean bookACar (BookACarDto bookACarDto);

    CarDto getCarById(Long carId);

    List<BookACarDto> getBookingsByUserId(Long userId);

    CarDtoListDto searchCar(SearchCarDto searchCarDto);

}

File: FrontEnd/car_rental_angular/src/app/modules/customer/service/customer.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from 'src/app/auth/services/storage/storage.service';

const BASIC_URL = ["http://localhost:8080"];

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http:HttpClient) { }

  getAllCars():Observable<any>{
    return this.http.get(BASIC_URL + "/api/customer/cars",{
      headers: this.createAuthorizationHeader()
    })
  }

  getCarById(carId : number):Observable<any>{
    return this.http.get(BASIC_URL + "/api/customer/car/"+carId,{
      headers: this.createAuthorizationHeader()
    })
  }

  bookACar(bookACarDto: any):Observable<any>{
    return this.http.post(BASIC_URL + "/api/customer/car/book",bookACarDto,{
      headers: this.createAuthorizationHeader()
    })
  }

  getBookingsByUserId():Observable<any>{
    return this.http.get(BASIC_URL + "/api/customer/car/bookings/"+StorageService.getUserId(),{
      headers: this.createAuthorizationHeader()
    })
  }

  searchCar(searchCarDto:any):Observable<any>{
    return this.http.post(BASIC_URL+"/api/customer/car/search",searchCarDto,{
       headers:this.createAuthorizationHeader()
    });
  }

  createAuthorizationHeader():HttpHeaders{
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer '+StorageService.getToken()
    );
  }

}
